simulation/traffic_simulator/src/entity/entity_base.cpp

```diff
@@ -721,8 +723,52 @@ void EntityBase::setTrafficLightManager(
   traffic_light_manager_ = traffic_light_manager;
 }
 
+void EntityBase::activateOutOfRangeJob(
+  double min_velocity, double max_velocity, double min_acceleration, double max_acceleration,
+  double min_jerk, double max_jerk)
+{
+  /**
+   * @brief This value was determined heuristically rather than for
+   * technical reasons.
+   */
+  constexpr double tolerance = 0.01;
+  job_list_.append(
+    /**
+     * @brief Checking if the values of velocity, acceleration and jerk are within the acceptable
+     * range
+     */
+    [this, tolerance, max_velocity, min_velocity, min_acceleration, max_acceleration, min_jerk,
+     max_jerk](double) {
+      auto velocity_ = status_.action_status.twist.linear.x;
+      auto accel_ = status_.action_status.accel.linear.x;
+      auto jerk_ = status_.action_status.linear_jerk;
+      if (!(min_velocity <= velocity_ + tolerance && velocity_ - tolerance <= max_velocity)) {
+        THROW_SPECIFICATION_VIOLATION(
+          "Entity: ", name, " - current velocity (which is ", velocity_,
+          ") is out of range (which is [", min_velocity, ", ", max_velocity, "])");
+      }
+      if (!(min_acceleration <= accel_ + tolerance && accel_ - tolerance <= max_acceleration)) {
+        THROW_SPECIFICATION_VIOLATION(
+          "Entity: ", name, " - current acceleration (which is ", accel_,
+          ") is out of range (which is [", min_acceleration, ", ", max_acceleration, "])");
+      }
+      if (!(min_jerk <= jerk_ + tolerance && jerk_ - tolerance <= max_jerk)) {
+        THROW_SPECIFICATION_VIOLATION(
+          "Entity: ", name, " - current jerk (which is ", jerk_, ") is out of range (which is [",
+          min_jerk, ", ", max_jerk, "])");
+      }
+      return false;
+    },
+    /**
+     * @brief This job is always ACTIVE
+     */
+    [this]() {}, job::Type::OUT_OF_RANGE, true, job::Event::POST_UPDATE);
+}
+
 auto EntityBase::setVelocityLimit(double) -> void {}
 
+auto EntityBase::setJerkLimit(double) -> void {}
```

Comment:
This function does not appear to do anything.
What is the purpose of this function?

Reply:
This is related to an earlier solution where the `max_jerk` was set by [this call](https://github.com/tier4/scenario_simulator_v2/blob/e504f6fad626a32e050f5dc4aa1f15d0a027f2ec/openscenario/openscenario_interpreter/include/openscenario_interpreter/simulator_core.hpp#L236-L238).
As you can see now `max_jerk` is passed in [activation](https://github.com/tier4/scenario_simulator_v2/blob/e504f6fad626a32e050f5dc4aa1f15d0a027f2ec/openscenario/openscenario_interpreter/include/openscenario_interpreter/simulator_core.hpp#L391-L394) so now `EntityBase::setJerkLimit()` do nothing - but I've decided to leave it as I assume it will be used in the future.

I asked about it [here](https://tier4.atlassian.net/browse/AJD-696?focusedCommentId=117435):
> If I understand correctly, it should have an effect and calling these methods [here](https://github.com/tier4/scenario_simulator_v2/blob/6c0b8b664a4e650c9b5039d4d9f924e286470624/openscenario/openscenario_interpreter/include/openscenario_interpreter/simulator_core.hpp#L236-L242) should do [it](https://github.com/tier4/scenario_simulator_v2/blob/8215ade14ca5ad3cfa3c025f2d2fb5797884f150/simulation/traffic_simulator/src/entity/entity_base.cpp#L766-L778). ([commit](https://github.com/tier4/scenario_simulator_v2/commit/8215ade14ca5ad3cfa3c025f2d2fb5797884f150))

You answered:

> Yes, it should affect.
> Your commit is looks good to me, thanks.

Since the definition of `EntityBase::setVelocityLimit` was empty before my earlier changes, I left `EntityBase::setJerkLimit ` empty as well.

Reply:
I think this function it was necessary, but unnecessary now.
Can you remove this function?

Reply:
Yes of course - I've removed [setJerkLimit](https://github.com/tier4/scenario_simulator_v2/pull/979/commits/034d844104e3b78416cf83de0ff2e061e648c2e9)

Reply:
Thanks!